### PR TITLE
Exit with real exit code from provision.sh

### PR DESCRIPTION
**Pull request by feiskyer**
Fix the following issues of CSE:

* `cmd && PROVISION_CODE=$?` made `PROVISION_CODE` always empty on cmd failure
* `$PROVISION_CODE` is also empty as it is parsed before running cse_cmd.sh

This PR ensures /opt/azure/containers/provision.sh is the last command and removes systemctl command.

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/cse_cmd.sh [hunk 1]
```diff
@@ -64,7 +64,7 @@ API_SERVER_NAME={{GetKubernetesEndpoint}}
 IS_VHD={{GetVariable "isVHD"}}
 GPU_NODE={{GetVariable "gpuNode"}}
 SGX_NODE={{GetVariable "sgxNode"}}
-AUDITD_ENABLED={{GetVariable "auditdEnabled"}} 
+AUDITD_ENABLED={{GetVariable "auditdEnabled"}}
 CONFIG_GPU_DRIVER_IF_NEEDED={{GetVariable "configGPUDriverIfNeeded"}}
 ENABLE_GPU_DEVICE_PLUGIN_IF_NEEDED={{GetVariable "enableGPUDevicePluginIfNeeded"}}
-/usr/bin/nohup /bin/bash -c "/bin/bash /opt/azure/containers/provision.sh >> /var/log/azure/cluster-provision.log 2>&1 && PROVISION_CODE=$?; systemctl --no-pager -l status kubelet 2>&1 | head -n 100; exit ${PROVISION_CODE}"
```
Comment (bowang-666) [inline parts/linux/cloud-init/artifacts/cse_cmd.sh:70]: what about change to "|| PROVISION_CODE=$?"

Comment (bowang-666) [reply parts/linux/cloud-init/artifacts/cse_cmd.sh:70]: we want to have kubelet status for debugging purpose

Comment (xuto2) [reply parts/linux/cloud-init/artifacts/cse_cmd.sh:70]: yes we need that too. But the && might be an issue, maybe we just do cmd; provision_code=$?

Comment (feiskyer) [reply parts/linux/cloud-init/artifacts/cse_cmd.sh:70]: `$PROVISION_CODE` would be parsed before running, so if we want to do more things, we need to move all commands into a separate script file.

Comment (feiskyer) [reply parts/linux/cloud-init/artifacts/cse_cmd.sh:70]: for debugging, we already have provisioning logs and exit code, I think that should be enough?

**Pull request merged**